app/models/digital_object/base.rb

```diff
@@ -77,5 +77,9 @@ def initialize
     def project_ids
       projects.each.map(&:id)
     end
+
+    def parents
+      parent_uids.map { |i| DigitalObject::Base.find(i) }
```

Comment:
Maybe good to cache this?
```
@parents ||= parent_uids.map { |i| DigitalObject::Base.find(i) }
```

Or actually, to avoid stale data and also to use cache when possible, maybe something like this:
```
if @parents
  @parents = @parents.map do |p|
    if p.updated_at != DigitalObject.get_updated_at(p.uid)
        DigitalObject::Base.find(p.uid)
    else
        p
    end
  end
else
  @parents ||= parent_uids.map { |i| DigitalObject::Base.find(i) }
end
```
Note that `DigitalObject.get_updated_at` doesn't exist yet, but could query solr.

Reply:
I was on the fence of adding an instance variable because I wasn't sure about how it would affect the other instance variables that are created. I can definitely add your first suggestion tho.

Reply:
Great, thanks!